pkg/keyidentifier/keyidentifier_test.go

```diff
@@ -0,0 +1,134 @@
+package keyidentifier
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/kolide/kit/logutil"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+// TestIdentifyFiles walks the testdata directory, and tests each
+// file. The expected results are extrapolated from the directory path
+func TestIdentifyFiles(t *testing.T) {
+	kIdentifer, err := New(WithLogger(logutil.NewCLILogger(true)))
+	require.NoError(t, err)
+
+	testFiles := []string{}
+
+	err = filepath.Walk("testdata", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return errors.Wrap(err, "failure to access path in filepath.Walk")
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		// This automatic tester can only parse things if they're stored
+		// in a folder hierarchy. So skip the odd test data
+		if strings.HasPrefix(path, "testdata/plaintext") || strings.HasPrefix(path, "testdata/encrypted") {
+			testFiles = append(testFiles, path)
+			return nil
+		}
+
+		return nil
+	})
+	require.NoError(t, err, "filepath.Walk")
+
+	for _, path := range testFiles {
+		testIdentifyFile(t, kIdentifer, path)
+	}
+}
+
+func testIdentifyFile(t *testing.T, kIdentifer *KeyIdentifier, path string) {
+	var err error
+	pathComponents := strings.Split(path, "/")
+	expected := &KeyInfo{
+		Format: pathComponents[4],
+	}
+
```

Comment:
Exactly. It might be too clever, but I was trying to make it simple to create and manage them all. At some point, I might convert to something more explicit.